runtime/src/iree/hal/drivers/hip/native_executable.c

```diff
@@ -255,136 +266,176 @@ iree_status_t iree_hal_hip_native_executable_create(
+              iree_hal_hip_ExportDef_module_ordinal_get(export_def);
+          hipModule_t module = per_device_data->modules[module_ordinal];
+          flatbuffers_string_t kernel_name =
+              iree_hal_hip_ExportDef_kernel_name_get(export_def);
+          hipFunction_t function = NULL;
+          status = IREE_HIP_CALL_TO_STATUS(
+              symbols, hipModuleGetFunction(&function, module, kernel_name),
+              "hipModuleGetFunction");
+          if (!iree_status_is_ok(status)) break;
+          if (!function) {
+            status = iree_make_status(IREE_STATUS_NOT_FOUND,
+                                      "exports[%" PRIhsz
+                                      "] kernel `%s` not found in modules[%u]",
+                                      i, kernel_name, module_ordinal);
+            break;
+          }
+
+          uint32_t block_shared_memory_size =
+              iree_hal_hip_ExportDef_block_shared_memory_size_get(export_def);
+          status = IREE_HIP_CALL_TO_STATUS(
+              symbols,
+              hipFuncSetAttribute(
+                  function,
+                  (hipFuncAttribute)
+                      HIP_FUNC_ATTRIBUTE_MAX_DYNAMIC_SHARED_SIZE_BYTES,
+                  block_shared_memory_size),
+              "hipFuncSetAttribute");
+          if (!iree_status_is_ok(status)) break;
+
+          // Package required parameters for kernel launches for each entry
+          // point.
+          iree_hal_hip_kernel_params_t* kernel_info =
+              &per_device_data->exports[i];
+          kernel_info->function = function;
+          const iree_hal_hip_BlockDims_t* block_dims =
+              iree_hal_hip_ExportDef_block_dims_get(export_def);
+          kernel_info->block_dims[0] = block_dims->x;
+          kernel_info->block_dims[1] = block_dims->y;
+          kernel_info->block_dims[2] = block_dims->z;
+          kernel_info->block_shared_memory_size =
+              iree_hal_hip_ExportDef_block_shared_memory_size_get(export_def);
+          kernel_info->constant_count =
+              iree_hal_hip_ExportDef_constant_count_get(export_def);
+          iree_hal_hip_BindingBits_vec_t binding_flags_vec =
+              iree_hal_hip_ExportDef_binding_flags_get(export_def);
+          kernel_info->binding_count =
+              iree_hal_hip_BindingBits_vec_len(binding_flags_vec);
+
+          IREE_TRACE({
+            iree_hal_debug_export_info_t* export_info =
+                (iree_hal_debug_export_info_t*)export_info_ptr;
+            export_info_ptr += iree_hal_debug_copy_export_info(
+                iree_hal_hip_ExportDef_debug_info_get(export_def), export_info);
+            kernel_info->debug_info.function_name = export_info->function_name;
+            kernel_info->debug_info.source_filename =
+                export_info->source_filename;
+            kernel_info->debug_info.source_line = export_info->source_line;
+          });
+        }
+      }
     }
   }
 
```

Comment:
extra nesting not required (it's not required and makes it harder to track the loop with the extra braces)

```suggestion
    IREE_IGNORE_ERROR(IREE_HIP_CALL_TO_STATUS(symbols, hipCtxPopCurrent(NULL),
```